docs/language-server/DocumentEventSyncPublisherSubscriberDesign.md

```diff
@@ -0,0 +1,114 @@
+# Document event sync publisher subscriber
+
+## Overview
+
+This document contains information about the implementation of a notification system inside the language server using the 
+PubSub architecture. Goal is to allow communication between language server and extensions synchronously.
+
+Following diagram shows how this has been implemented. 
+
+![alt text](../images/language-server/001-DocumentEventPubSub-class-diagram.png)
+
+- Event Subscribers and Event Publishers get loaded using JavaSPI and get subscribed to publishers when a document 
```

Comment:
```suggestion
- Event Subscribers and Event Publishers get loaded using JavaSPI and subscribed to an EventKind. A subscriber is notified when a publisher publishes an event (document update, open, close, etc) under that EventKind 
```

Reply:
For now, we are not publishing events on `didClose` and we do not have to publish events on `didClose` right? @IMS94

Reply:
Can you be more specific on the `image` folder move? currently the image is in the `docs/images/language-server/001-DocumentEventPubSub-class-diagram.png` path

Reply:
Let's add LS related images in `docs/language-server/images` folder. We have to create that.

Reply:
what about the `didClose` @IMS94

Reply:
Though we don't support `didClose` as an event, it's a possible extension in the future. Just wanted the reader to get the idea of event kinds and subscribing.